opslevel/resource_opslevel_integration_google_cloud.go

```diff
@@ -0,0 +1,252 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &integrationGoogleCloudResource{}
+	_ resource.ResourceWithImportState = &integrationGoogleCloudResource{}
+)
+
+func NewIntegrationGoogleCloudResource() resource.Resource {
+	return &integrationGoogleCloudResource{}
+}
+
+type integrationGoogleCloudResource struct {
+	CommonResourceClient
+}
+
+type integrationGoogleCloudResourceModel struct {
+	Aliases               types.List   `tfsdk:"aliases"`
+	ClientEmail           types.String `tfsdk:"client_email"`
+	CreatedAt             types.String `tfsdk:"created_at"`
+	Id                    types.String `tfsdk:"id"`
+	InstalledAt           types.String `tfsdk:"installed_at"`
+	Name                  types.String `tfsdk:"name"`
+	OwnershipTagKeys      types.Set    `tfsdk:"ownership_tag_keys"`
+	PrivateKey            types.String `tfsdk:"private_key"`
+	Projects              types.List   `tfsdk:"projects"`
+	TagsOverrideOwnership types.Bool   `tfsdk:"ownership_tag_overrides"`
+}
+
+func newIntegrationGoogleCloudResourceModel(googleCloudIntegration opslevel.Integration, givenModel integrationGoogleCloudResourceModel) integrationGoogleCloudResourceModel {
+	resourceModel := integrationGoogleCloudResourceModel{
+		Aliases:     OptionalStringListValue(googleCloudIntegration.GoogleCloudIntegrationFragment.Aliases),
+		ClientEmail: givenModel.ClientEmail,
+		CreatedAt:   ComputedStringValue(googleCloudIntegration.CreatedAt.Local().Format(time.RFC850)),
```

Comment:
Why `RFC850` instead of `RFC3339` ?

Reply:
I ripped this from the Azure PR for Terraform since I wanted them both to be consistent. https://github.com/OpsLevel/terraform-provider-opslevel/pull/420/files#diff-62f5fc78cb3be71c03ed02fbd278cd3bf88c8a8f7012d259a314ae9e08828286

Side note I checked what we use for the AWS integration in TF but it turns out we don't show the created/installed date at all. https://github.com/OpsLevel/terraform-provider-opslevel/blob/87183cbec8e9bbb5346b16632a13fcf19e7ac78e/opslevel/resource_opslevel_integration_aws.go

Reply:
hmm - we should use RFC3339 because thats closer to ISO8601 that our API uses for timestamps and we should make it UTC time because then it won't change just because someone ran plan/apply on a computer in CT vs PT vs ET.

Reply:
@rocktavious instead of introducing a breaking change, for the existing AWS integration and Azure integration and this GCP integration we could introduce 2 additional fields 

`CreatedAtUTC` and `InstalledAtUTC` - what do you think?

Reply:
These are computed fields - so we can change how the state store them from the API response.  Please do not add new fields.

Reply:
@rocktavious changed the field for the GCP integration resource to use UTC and RFC3339.